### PR TITLE
3218 make the function fetchworkspacememberthreads idempotent

### LINKED ISSUE #3218: Make the function fetchWorkspaceMemberThreads idempotent
## Scope & Context

We need the function `fetchWorkspaceMemberThreads` to be idempotent so we can retry jobs when they fail.

## Technical inputs

Don't fetch the messages if they are already in the database.
- Get the messages from the workspaceMember
- Get the messageIds from gmail
- Only fetch the messages which are not already in the db.

### PR DESCRIPTION
**Pull request by bosiraphael**
Closes #3218

### CODE REVIEW

File: packages/twenty-server/src/workspace/messaging/services/fetch-workspace-messages.service.ts [hunk 1]
```diff
@@ -224,4 +231,74 @@ export class FetchWorkspaceMessagesService {
...
+      LEFT JOIN ${dataSourceMetadata.schema}."messageThread" ON message."messageThreadId" = ${dataSourceMetadata.schema}."messageThread"."id"
+      LEFT JOIN ${dataSourceMetadata.schema}."messageChannel" ON ${dataSourceMetadata.schema}."messageThread"."messageChannelId" = ${dataSourceMetadata.schema}."messageChannel"."id"
+      WHERE ${dataSourceMetadata.schema}."messageChannel"."connectedAccountId" = $1`,
+        [connectedAccountId],
+      );
+
+    return messageIds.map((message) => message.externalId);
+  }
+
+  async getAllSavedMessageThreadsIdsForConnectedAccount(
```
Comment (Weiko) [inline packages/twenty-server/src/workspace/messaging/services/fetch-workspace-messages.service.ts:252]: We can probably get both message and thread external IDs in a single request?

File: packages/eslint-plugin-twenty/dist/index.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
```
Comment (Weiko) [inline packages/eslint-plugin-twenty/dist/index.js:2]: Probably not intended to push all of this?

**Review by Weiko — COMMENTED** [review]
Strategy lgtm, we can improve later to delegate this to psql once we can add unicity constraints to standard objects. 
Left a comment to address then let's merge it 💪@bosiraphael

File: packages/twenty-server/src/workspace/messaging/services/fetch-workspace-messages.service.ts (hunk 1 above)
Comment (bosiraphael) [reply packages/twenty-server/src/workspace/messaging/services/fetch-workspace-messages.service.ts:252]: Right !

File: packages/eslint-plugin-twenty/dist/index.js (hunk 2 above)
Comment (bosiraphael) [reply packages/eslint-plugin-twenty/dist/index.js:2]: 👍

**Review by Weiko — APPROVED** [review]
LGTM!

**Pull request merged**